ksql-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlClient.java

```diff
@@ -38,37 +44,86 @@
     JsonMapper.INSTANCE.mapper.registerModule(new KsqlTypesDeserializationModule(false));
   }
 
-  private final Client httpClient;
+  public static final String DISABLE_HOSTNAME_VERIFICATION_PROP_NAME
+      = "ksql.client.disable.hostname.verification";
+  public static final String TLS_ENABLED_PROP_NAME = "ksql.client.enable.tls";
+
+  private final Vertx vertx;
+  private final HttpClient httpClient;
   private final LocalProperties localProperties;
+  private final Optional<String> basicAuthHeader;
+  private final boolean isTls;
 
   public KsqlClient(
       final Map<String, String> clientProps,
       final Optional<BasicCredentials> credentials,
-      final LocalProperties localProperties
+      final LocalProperties localProperties,
+      final HttpClientOptions httpClientOptions
   ) {
-    this(HttpClientBuilder.buildClient(clientProps), credentials, localProperties);
+    this.localProperties = requireNonNull(localProperties, "localProperties");
+    this.basicAuthHeader = createBasicAuthHeader(
+        Objects.requireNonNull(credentials, "credentials"));
+    this.vertx = Vertx.vertx();
+    if ("true".equals(clientProps.get(DISABLE_HOSTNAME_VERIFICATION_PROP_NAME))) {
+      httpClientOptions.setVerifyHost(false);
+    }
+    if ("true".equals(clientProps.get(TLS_ENABLED_PROP_NAME))) {
+      httpClientOptions.setSsl(true);
+      isTls = true;
+    } else {
+      isTls = false;
+    }
+    final String trustStoreLocation = clientProps.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
+    if (trustStoreLocation != null) {
+      httpClientOptions.setTrustStoreOptions(new JksOptions().setPath(trustStoreLocation)
+          .setPassword(clientProps.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG)));
+      final String keyStoreLocation = clientProps.get(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG);
+      if (keyStoreLocation != null) {
+        httpClientOptions.setKeyStoreOptions(new JksOptions().setPath(keyStoreLocation)
+            .setPassword(clientProps.get(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG)));
+      }
+    }
+    this.httpClient = vertx.createHttpClient(httpClientOptions);
   }
 
   @VisibleForTesting
   KsqlClient(
```

Comment:
Can we have the other constructor call this one, in order to minimize divergence between tests and production code?

Reply:
ack